tools/build-pypi-artifacts.py

```diff
@@ -0,0 +1,253 @@
+    status, _ = proc.communicate()
+    status = status.decode("ascii", "replace")
+    if not "HEAD is now at" in status:
+        print(f"{status}\n")
+        raise RuntimeError("Resetting to original commit may have gone wrong")
+
+
+def generate_sdist():
+    proc = subprocess.run([sys.executable, '-m', 'build', '--sdist',
+                           '--no-isolation', '--skip-dependency-check'])
+    distdir = repo_root_dir / 'dist'
+    if not len(list(distdir.glob('scipy-*.tar.gz'))) == 1:
+        raise RuntimeError("sdist creating not succesful!")
+
+
+def create_sdist_for_pypi():
+    check_submodules()
+    check_not_dirty()
+    commit_id = get_current_commit_id()
+
+    concat_license_files()
+    modify_pyproject_toml()
+
+    commit_changed_files()
+    generate_sdist()
+    reset_to_previous_commit(commit_id)
+
+
+def create_wheel_for_pypi(skip_check=False):
+    """
+    Create a wheel with the appropriate wheel-specific license info (for
+    vendored libraries) and `_distributor_init.py` file.
+    """
+    # TODO: adapt for what is going on in https://github.com/MacPython/scipy-wheels
+    #       - grab the correct build of OpenBLAS (or keep separate?)
+    #       - patch_code.sh changes need to live here (double check and
+    #         document what mesonpy does with vendoring)
+    #       - patch_numpy.sh isn't needed
+    #       - set MACOSX_DEPLOYMENT_TARGET in environment here
+    #       - unclear whether we have a use for setting PYTHONFAULTHANDLER=1
+    #       - verify_init.py is likely good to incorporate here
+    raise NotImplementedError('TODO')
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description=textwrap.indent(
+            textwrap.dedent(
+                """
+                Tool to build SciPy release artifacts (sdist, wheels) for PyPI.
+
+                This tool is needed for releases because we need to take
+                PyPI-specific actions like amend license files and
+                the numpy dependency in pyproject.toml before invoking a
+                standard tool like `pypa/build`.
+
+                Usage:
+
+                  $ python build-pypi-artifacts.py            # build sdist, then a wheel
+                  $ python build-pypi-artifacts.py --sdist    # build sdist only
+                  $ python build-pypi-artifacts.py --wheel path/to/sdist
+
+                Note that the sdist can be created in any environment, there
+                is no platform or Python-version specific content that ends up
+                in the sdist. The wheels on the other hand must be created in
+                an environment that has the correct dependencies already
+                installed. If there's a mismatch in, for example, the version
+                of numpy that is installed in the (CI) environment compared to
+                what's specified in `pyproject.toml`, an error will be raised.
```

Comment:
There will be only one way. Look at this like this:
1. `cibuildwheel` (or the current `scipy-wheels` repo based on `multibuild`) are meant to (a) set up the environments needed build wheels for PyPI, and (b) provide a means to trigger those builds.
2. Both `cibuildwheel` and `multibuild` need to run _one single command_, e.g. `pip wheel` or `python -m build`, to actually build the wheel after all of the environment setup (and maybe applying some patches). Currently that is `python setup.py bdist_wheel`. We need a replacement for that - which has to come from this PR.

Reply:
That said, I would like to have more of the logic of what is now in scipy-wheels/cibuildwheel CI config files to live in scripts, so that it can be tested and debugged locally rather than only in CI itself. That's always a good thing.